### ISSUE TITLE
New Feature: UI Flow Enhancement

### ISSUE THREAD

**Issue by cchunduri** [Tonterias/spingularchatsql6#12]
At Present, after user login, the chat window is coming. 

**Requirement:**

- On User Login, for the first time, he should be presented with the chat rooms that are available for him to join
- The user selects the Chat room, from the next logins he'll be directed to the same chat room
- We'll have options for him to leave chat-room
- On leaving the chat-room he'll get the list of chat-rooms available for him.
- The user can also create his own chat-rooms
- The user should not be able to see the other's chat rooms unless he has an invitation for those chat rooms

**Comment by Tonterias** [comment]
I think that is what we need. I would add that we need to explain things a little bit more (add some literature to it) and then this is the analysis that comes after explaining it. 

Let me give an example... We would like to make a chat in Jhipster where the user can create Chatrooms and other user can be allowed to join them as ChatRoomAllowedUsers... And the you do this part where you break the whole thing is pieces.

Do you know what I mean? In the post https://www.spingular.com/#/post/22/view
We need to explain the whole thing a little bit before we get into explaining the details for the BackLog. That way, anyone will be able to follow the programming.

Until now it was easy because we are only 3 and we all know what is the basic idea behind a Chat, but as the move forward we will need to explain things a little bit more and make detailed requirements. We can even use images so everything is clear for the programmers.

What do you think?

**Comment by Tonterias** [comment]
For example, are we sure about this: The user should not be able to see the other's chat rooms unless he has an invitation for those chat rooms.

In Gitter, you can see that there are ChatRooms about Jhipster if you search for them and then you can join them. How will they join it if they can not see it. 

I think we have to make a distinction between public and private ChatRooms, right? We have a field on our ChatRoom to do that....

**Comment by cchunduri** [comment]
> For example, are we sure about this?:
> 
> The user should not be able to see the other's chat rooms unless he has an invitation for those chat rooms.
> 
> In Gitter, you can see that there are ChatRooms about Jhipster if you search for them and then you can join them. How will they join it if they can not see it.
> 
> I think we have to make a distinction between public and private ChatRooms, right? We have a field on our ChatRoom to do that.... but we need the story telling so everyone can follow and find that may be we should let users search the ChatRooms, right?

Yes, Mike, this is making sense to me. We can have both public chat-rooms and private chat rooms. Where public chat rooms will be visible to everyone and private chat rooms will be visible to those who have an invitation.

**Comment by Tonterias** [comment]
The point is how do you explain something so other people can look at it (read it is see images) and find what is missing or a problem or a better way to do it before you start writing code?
